o1-labs/o1js-bindings
# Merge develop back to main

- Fixup bindings for hex serialization
- If recompiling the WASM changes it, something has gone very wrong
- bindings
- separate type checking from build script
- fixup
- move bindings location and add guarding d.ts
- Fixup to make npm run bindings happy
- tweak how merge conflict is resolved
- Run prettier on bindings.js
- wasm/{wasm_vector, plonk_proof?: move WasmVecVecF{p,q}
- wasm/src/index: define wasm lookup table
- wasm/src/index: define wasm runtime table cfg
- update bindings
- wasm/src/plonk_proof.rs: define wasm runtime tables
- kimchi/js/bindings.js: add runtime_tables to proof_create
- bindings.js: remove duplicated definition
- kimchi/js/bindings.js: define js/wasm converters for lookup tables
- bindings_js_test: add empty runtime table cfg arg
- wasm/index: add runtime table cfgs arg
- bindings.js: add runtime_table_cfgs arg
- kimchi/js/bindings.js: define converters for runtime table cfg
- Run npm run bindings
- Update bindings
- Update bindings
- deduplicate logic of parsing override_wrap_domain
- Run prettier -w kimchi/js/bindings.js
- CI: add basic workflow to run prettier
- Run prettier -w kimchi/js/bindings.js
- Import prettier config from snarkyjs
- update worker spec
- kimchi/js/bindings.js: fix runtime table cfg field access
- kimchi/js/bindings.js: change converters for runtime, lookup table and runtime table cfg
- kimchi/js/bindings.js: unify field access of caml record
- Run npm run bindings
- Remove lookup tables converters
- Remove runtime table converters
- Remove caml_is_empty_array_caml_array and caml_create_rust_empty_vector
- Regenerate bindings with npm run bindings
- Run npm run bindings
- Add lookup tables in worker-spec + comments
- Add comments related to changes that must appear in worker-spec
- Reintroduce converters for lookup tables and add it in stubs
- binding_js_test.ml: add empty lookup tables
- index: add param lookup tables
- bindings.js: add bytecode version of the caml stubs
- Artefacts: run npm run bindings
- Add runtime tables to worker-spec and add comments
- plonk_proof.rs: pass runtime tables to create proof
- kimchi/js/bindings.js: add runtime_tables to proof_create
- bindings_js_test: add runtime tables
- snarky_bindings: pass empty runtime tables atm
- Reintroduce JS converters for runtime tables
- Artefacts: generate using npm run bindings
- Add js/**/*.js in prettier check
- Run the following command:
- CI: run prettier check on all JS files in kimchi/js
- Replace joo_global_object by globalThis
- Regenerate bindings with npm run bindings
- prettier
- Introduce different Wasm equivalent types and add lookup verif index
- bindings.js: add to/of_rust conversions of new structures
- Update artefacts after run npm run bindings
- Expose lookup commitments
- bindings.js: removing unused parameter
- bindings.js: remove TODO
- bindings.js: use caml_opt_to_rust/of_rust
- Update bindings after artefacts generation
- update bindings
- address feedback - add documentation in comments
- Remove runtime tables parameter
- Run npm run bindings to generate bindings artefacts
- Removing runtime table parameters in binding_js_test.ml
- update bindings
- Update bindings_js_test with new optional commitments
- wasm: update verifier index
- wasm/js: add support for new commitments in bindings
- Update compiled bindings
- Update bindings for additional public input evaluation
- Fixup bindings test
- Js bindings: remove trailing ,0
- Rerun npm run bindings
- Update generated bindings
- update bindings
- Update bindings for public_input_evals
- Update bindings for chunked zero-knowledge
- Update artefacts with npm run bindings
- bindings.js: run prettier
- Add missing OpeningProof type argument
- DlogVerifierIndex: reorganize struct fields
- Remove old struct field
- srs parameter is not an option anymore
- Import SRS as ISRS
- Partially reversed d40dd6c6a5517bbebd3c77abd54c2f9813724d7f
- DlogVerifierIndex: update srs field
- Run cargo fmt
- Update rust version to nightly release close to 1.72
- Cargo update
- Run prettifier
- Updating artefacts
- Bump up artefacts after updating to wasm-pack 0.12.1
- Strictly require wasm-bindgen 0.2.84
- Run cargo generate-lockfile
- Regenerate bindings artefacts
- Adding comment regarding the version of wasm-bindgen
- upgrade wasm-bindgen to 0.2.85
- Run cargo generate-lockfile
- Generate bindings artefacts with 0.2.85 wasm-bindgen
- Use __wbg_ptr instead of ptr
- Regenerate bindings after stopping using ptr attr
- web-backend.js: use __wbg_ptr instead of ptr
- Fixup compilation for WASM
- refactor(web-backend.js): rename 'init' to '__wbg_init'
- change .ptr in web backend and revert unnecessary change
- update bindings with all the right branches
- update verification evals
- add runtime table types
- runtime tables to rust
- lookup tbls & runtime table cfgs to rust
- fixup
- fixup
- fix build script
- updated wasm types
- fix __wbg_ptr
- update proof type
- update bindings
- run prettier
- revert useless changes
- remove TODO comments
- custom lint yaml
